### PR TITLE
Address spec issue #915 - Specify the types allowed for header stack indexes

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
@jnfoster Here is a proposed change for issue #915.

I have no strong preference for or against allowing serializable enum types as header stack indexes.  It seems reasonable to me to allow them, or not.

The rest seems fairly safe and conservative to me.

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -4111,6 +4111,17 @@ expressions are legal:
...
+  The `index` is an expression that must be one of the following types:
+
+  - `int` - an infinite-precision integer (section
+    [#sec-infinite-precision-integers])
+  - `bit<W>` - a `W`-bit unsigned integer where `W >= 1` (section
+    [#sec-unsigned-integers])
+  - `int<W>` - a `W`-bit signed integer where `W >= 1` (section
+    [#sec-signed-integers])
+  - a serializable `enum` with an underlying type that is `bit<W>` or
+    `int<W>` (section [#sec-enum-types]).
```
Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:4123]: @jafingerhut AI Andy to create a small test program and see whether p4c accepts a header stack index with type of serializable enum is supported today.

**Comment by mbudiu-vmw** [comment]
Check whether the implementation supports enums.
Probably this should be accepted.

File: p4-16/spec/P4-16-spec.mdk (hunk 1 above)
Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk:4123]: A simple test program shows that recent versions of p4c given an error when one attempts to use an expression of type `enum bit<8> ...` as the index of a header stack reference: https://github.com/p4lang/p4c/issues/2726

**Review by mbudiu-vmw — APPROVED** [review]
The compiler now supports all these cases: https://github.com/p4lang/p4c/pull/2738

**Comment by jafingerhut** [comment]
@mbudiu-vmw I am pretty sure we agreed to merge this into the spec during the 2021-May-03 LDWG meeting.  It had a conflict (in the change log section, as expected) which should now be resolved.  I will let you push the merge button for this, in case I am mixed up on PRs.

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**